edg/parts/Microcontroller_Stm32g431.py

```diff
@@ -0,0 +1,273 @@
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        # Power and ground
+        self.pwr = self.Port(VoltageSink(
+            voltage_limits=(1.71, 3.6) * Volt,
+            current_draw=(0.001, 150.0) * mAmp  # table 15
+        ), [Power])
+
+        self.gnd = self.Port(Ground(), [Common])
+
+        self.swd = self.Port(SwdTargetPort.empty())
+        self._io_ports.insert(0, self.swd)
+        self.nrst = self.Port(DigitalSink.empty(), optional=True)  # internally pulled up
+
+    def _system_pinmap(self) -> Dict[str, CircuitPort]:
+        return VariantPinRemapper({
+            'Vdd': self.pwr,
+            'Vss': self.gnd,
+            # 'VddA': self.pwr,
+            # 'VssA': self.gnd,
+            'BOOT0': self.gnd,
+            'PG10-NRST': self.nrst,
+        }).remap(self.SYSTEM_PIN_REMAP)
+
+    def _io_pinmap(self) -> PinMapUtil:
+        input_range = self.gnd.link().voltage.hull(self.pwr.link().voltage)
+        io_voltage_limit = (input_range + (-0.3, 3.6) * Volt).intersect(
+            self.gnd.link().voltage + (-0.3, 5.5) * Volt)  # Section 5.3.1
+        input_threshold_factor = (0.3, 0.7)  # Section 5.3.14
+        current_limits = (-20, 20) * mAmp  # Table 15
+
+        dio_ft_model = DigitalBidir.from_supply(
+            self.gnd, self.pwr,
+            voltage_limit_abs=io_voltage_limit,
+            current_limits=current_limits,
+            input_threshold_factor=input_threshold_factor,
+            pullup_capable=True, pulldown_capable=True
+        )
+        dio_tt_model = DigitalBidir.from_supply(
+            self.gnd, self.pwr,
+            voltage_limit_tolerance=(-0.3, 0.3) * Volt,  # Table 19
+            current_draw=(0, 0) * Amp, current_limits=current_limits,
+            input_threshold_factor=input_threshold_factor,
+            pullup_capable=True, pulldown_capable=True
+        )
+
+        # Pin definition, 4.10
+        dio_fta_model = dio_ftca_model = dio_ftf_model = dio_ftfa_model = \
+            dio_ftfu_model = dio_ftfd_model = dio_ftda_model = dio_ftu_model = dio_ft_model
+        dio_tta_model = dio_tt_model
+
+        dio_ftc_model = DigitalBidir.from_supply(
+            self.gnd, self.pwr,
+            voltage_limit_abs=(-0.3, 5.0) * Volt,
+            current_limits=current_limits,
+            input_threshold_factor=input_threshold_factor,
+            pullup_capable=True, pulldown_capable=True
+        )
+
+        adc_model = AnalogSink.from_supply(
+            self.gnd, self.pwr,
+            voltage_limit_tolerance=(-0.3, 0.3) * Volt,
+            signal_limit_tolerance=(0, 0),
+            impedance=(50, float('inf')) * kOhm  # TODO: this is maximum external input impedance, maybe restrictive
+        )
+        dac_model = AnalogSource.from_supply(
+            self.gnd, self.pwr,
+            # signal_out_bound=(0.2*Volt, -0.2*Volt), signal_out_bound only applies when output buffer on
```

Comment:
Dead code is generally discouraged style-wise. Consider just leaving the bound in there (as a safe bound) and having a comment that it only applies in a specific case.